app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb

```diff
@@ -7,12 +7,22 @@ module ValidateAndDecode
       # for the 2nd validation
       ValidateAndDecodeToken ||= CommandClass.new(
         dependencies: {
+          fetch_signing_key: ::Util::ConcurrencyLimitedCache.new(
```

Comment:
maybe we should change it to? `fetch_cached_signing_key` or `fetch_signing_key_from_cache` ?

Reply:
Changed to fetch_signing_key_from_cache